tests/test_delta_lake.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import chispa
+import pyspark
+from delta import *
+
+builder = (
+    pyspark.sql.SparkSession.builder.appName("MyApp")
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+    .config(
+        "spark.sql.catalog.spark_catalog",
+        "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+    )
+)
+
+spark = configure_spark_with_delta_pip(builder).getOrCreate()
+
+
+def test_reference_table1_a():
```

Comment:
I would like to see a more of a framework approach as i said in the doc... some way to automatically enumerate the list of the reference table and all of their information and run all of them ... so that as we add more reference tables here ... all the projects depending upon it can automatically pick those tables up without adding additional per-table unit test code like you are doing

Reply:
@tdas do you have in mind something like this?

https://github.com/delta-incubator/dat/pull/4

creating an object model to describe tables and use cases, and potentially be able to export it to json.

@MrPowers regarding tables, I am fine to have them under git as long as they are automatically committed by the piipeline and not by the user. I can help in setting a Github workflow that does it

Reply:
@edmondo1984 

- yeah this is exactly what i had in mind, the object model!! at the very least, the [base object model fields](https://github.com/delta-incubator/dat/pull/4/files#diff-b5e1446e2d4c16242c8fc7aeadb32490bc93570b8423a638fcad02b1c29c2505R9-R13). These fields will be the same (or superset) of the information in my suggestion metadata.json. So the framework will parse the metadata.json for each table and expose them as objects if this type. Note that this framework will be specific to each project that implements delta protocol. delta-spark will have this framework in java/python, delta-standalone in java, delta-flink in java, delta-rs in rust. so all the DAT project has to provide is a "reference implementation" of this framework. 

- i dont understand what you mean by tables being `automatically committed by the pipeline`? maybe i am missing some context.

- your project seems to have a lot boilerplate files.... that looks scary to me.

Reply:
on second thought... the "reference" framework can actually be more than just a reference... it should be used to validate that the added reference tables satisfy the ref table specs... has correct metadata in expected parseable form, has the expected answer is parquet/json/some-parseable format, etc.

Reply:
Thanks, let me comment to the second and third point:

Point 2) Auto-commit of assets
Normally assets that can be generated by the code are not committed manually under VCS, because they could be easily de-sync from the source code. It's kind of committing a compiled object file for C, or a .class or even a jar in the Java world.  Other examples: classes generated by compiler plugins looking for example at XML schemas or Avro schemas are never committed to VCS but are included in Jars when they are built and published to maven central.

In our case it would mean that the table data should not be committed in Git manually, so we have two options:
- use GitHub workflows to generate the table on each commit and then commit them (this is a common practice)
- use GitHub workflows to commit the generated tables on a separate Git Repo

Point 3) Other "files" in the repo

Besides python code, my PR contains:
- a pull request template so to get us to follow a structure when collaborating
- an automated pipeline that will verify code adheres to standard on each commit and tests pass
- a .gitignore
- poetry.lock and pyproject.toml, files used by poetry as a dependency manager

Which of these files do you suggest we remove?

Reply:
>  some way to automatically enumerate the list of the reference table and all of their information and run all of them ... so that as we add more reference tables here ... all the projects depending upon it can automatically pick those tables up without adding additional per-table unit test code like you are doing

My original understanding is that we had a standardized directory structure, so implementations would simply `ls` the `reference_tables` folder, then run their tests on tests on each folder. Is something more complicated necessary?